functions/src/api/llama.api.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * Client handling communications with an Ollama server.
+ * 
+ * Code assumes that the LLaMa instance is hosted on the provided IP address,
+ * and is managed through the `ollama` framework (https://github.com/ollama/ollama).
+ * Example docker instance hosting an ollama server: https://github.com/dimits-ts/deliberate-lab-utils/tree/master/llm_server
+ * 
+ * Note: there already exists a client library for JavaScript, but not for Typescript.
```

Comment:
I'm working on adding some more API support, and I'm planning to support Ollama either via the openai library and Ollama's openai API compatibility, or via multi-llm-ts. This looks good in the meantime, just a heads up, and I wouldn't recommend putting a lot more work into polishing this.

Reply:
The openai API will work fine wherever ollama is hosted. That method won't touch openai's servers or anything, it's just a client library that we don't need to maintain ourselves.